internal/server/handlers.go

```diff
@@ -85,7 +85,6 @@ func (a *API) CreateAccessKey(c *gin.Context, r *api.CreateAccessKeyRequest) (*a
 	accessKey := &models.AccessKey{
 		IssuedFor:         r.UserID,
 		Name:              r.Name,
-		ProviderID:        access.InfraProvider(c).ID,
```

Comment:
We still need this or running `infra tokens add` will fail when used with directly created access keys

Reply:
The behaviour is still the same, it's being set in `access.CreateAccessKey` which is called right after this.  Sorry I didn't call this out in the description.

`data.CreateAccessKey` validates that `ProviderID` is not 0, and we have tests for the `CreateAccessKey` endpoint which should cover this behaviour.